### PR TITLE
feat: dark mode toast

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ksdark-mode-toast-hashicorp.vercel.app/packer) 🔎
- [Asana task](https://app.asana.com/0/1203652232454021/1204260876693590) 🎟️

## 🗒️ What

Adds a toast to notify the user of theme support and where to access the theme switcher. 
Screenshot 2023-04-03 at 1 04 49 PM
Screenshot 2023-04-03 at 1 04 44 PM

## 🧪 Testing

- [Visit the preview](https://dev-portal-git-ksdark-mode-toast-hashicorp.vercel.app/packer)
- Validate that the toast renders. 
- If you dismiss and refresh or start a new session, validate that it does not render again
- Test in both dark and light mode
- The toast should not render on the home or sign-up pages

### CODE REVIEW

File: src/lib/toast/make-dark-mode-notification.ts [hunk 1]
```diff
@@ -39,8 +39,9 @@ export function makeWelcomeToast() {
 	// If welcome toast cookie has not expired and all other checks have passed, render toast
 	toast({
 		color: ToastColor.highlight,
-		title: `Welcome to HashiCorp Developer!`,
-		description: 'Your destination for documentation and tutorials.',
+		title: `Theme support is now available!`,
+		description:
```
Comment (kendallstrautman) [inline src/lib/toast/make-dark-mode-notification.ts:43]: Still working on the copy.

File: src/layouts/core-dev-dot-layout/index.tsx [hunk 2]
```diff
@@ -20,11 +20,15 @@ const CoreDevDotLayout = ({ children }: CoreDevDotLayoutProps) => {
...
+	const isToastPath =
+		pathname !== '/' &&
+		pathname !== '/sign-up' &&
+		pathname !== '/_error' &&
+		!isSwingset
 
 	useEffect(() => {
-		if (isReady && isToastPath) {
-			makeWelcomeToast()
+		if (__config.flags.enable_dark_mode && isReady && isToastPath) {
```
Comment (kendallstrautman) [inline src/layouts/core-dev-dot-layout/index.tsx:30]: Don't need a flag if we merge this into the assembly

File: src/layouts/core-dev-dot-layout/index.tsx [hunk 3]
```diff
@@ -20,11 +20,15 @@ const CoreDevDotLayout = ({ children }: CoreDevDotLayoutProps) => {
 	const { asPath, pathname, isReady } = router
 
 	const isSwingset = asPath.startsWith('/swingset')
-	const isToastPath = pathname !== '/' && pathname !== '/_error' && !isSwingset
+	const isToastPath =
+		pathname !== '/' &&
+		pathname !== '/sign-up' &&
```
Comment (kendallstrautman) [inline src/layouts/core-dev-dot-layout/index.tsx:25]: Will refactor this to use config in a follow up task https://app.asana.com/0/1203652232454021/1204323512217789

**Review by zchsh — APPROVED** [review]
✅ Followed the testing steps - all seems great to me! 🖤 Saw the toast, but not on home page or sign-up, and dismiss worked 👍

**Review by braydenlove — APPROVED** [review]
Awesome job.

**Pull request merged**